### ISSUE TITLE
Using parallel_split_test, seeing Errno::ENOENT: No such file or directory - /usr/sbin/sysctl

### ISSUE THREAD

**Issue by pamgluss-slack** [grosser/parallel#300]
Error we're getting:
`bundle exec parallel_split_test spec --tag ~unstable --tag ~skip_regression
17:29:50 bundler: failed to load command: parallel_split_test (/mnt/jenkinsslave/workspace/webapp-regression-api/tests/rspec-api-tests/.rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/bin/parallel_split_test)
17:29:50 Errno::ENOENT: No such file or directory - /usr/sbin/sysctl`

The time we started seeing this was 3:30 PM on our Jenkins builds. This lines up with the [release of 1.20.2](https://github.com/grosser/parallel/releases/tag/v1.20.2). A workaround is for us to use 1.20.1 but I wanted to bring this to yalls attention.

Ruby version 2.4.1
Rspec Version 3.9.0

**Comment by grosser** [comment]
thx
comes from https://github.com/grosser/parallel/blob/f456129a93bcafd50e9f2bef70030a6cc945b219/lib/parallel/processor_count.rb#L16

are you running on mac ?
what does `/usr/sbin/sysctl -n hw.physicalcpu` return ?

**Comment by pedromartinez** [comment]
```bash
pedro@pedros-mac-mini ~ % /usr/sbin/sysctl -n hw.physicalcpu
6
```

**Comment by pedromartinez** [comment]
I'll open a PR to revert.

**Comment by pedromartinez** [comment]
https://github.com/grosser/parallel/pull/301

**Comment by pedromartinez** [comment]
Note:

`physical_processor_count` appears to be used in the `parallel_split_test` gem but not in `parallel_tests`.

Even after my PR, parallel tests still returned 12 for processor_count.

**Comment by grosser** [comment]
@pamgluss-slack can you try and find out why sysctl is missing for you ?

**Comment by grosser** [comment]
it uses processor_count https://github.com/grosser/parallel_tests/blob/master/lib/parallel_tests.rb#L20
that seems bad since either one or the other is better for tests, so if you can find out if 6/12 is best then we can change one or the other

**Comment by grosser** [comment]
let's revert for now since it looks like we need a different fix

**Comment by pamgluss-slack** [comment]
I can get details for you guys in the morning PST, but I believe Jenkins runs Linux machines

**Comment by pedromartinez** [comment]
@grosser I agree, it should probably use the same method for both libraries.

Here are two different work projects.

A smaller project:

6x:

```
426 examples, 0 failures
Took 50 seconds

12x:

```
426 examples, 0 failures
Took 59 seconds
```

and

A larger project:

6x:

```
8949 examples, 0 failures, 4 pendings
Took 406 seconds (6:46)
```

12x:

```
8949 examples, 0 failures, 4 pendings
Took 332 seconds (5:32)
```

**Comment by pedromartinez** [comment]
For reference:

```bash
pedro@pedros-mac-mini % sysctl -n hw.physicalcpu
6

pedro@pedros-mac-mini  % sysctl -n hw.ncpu       
12
```

**Comment by grosser** [comment]
seems to be unclear, so I'd rather use 12 since changing parallel_tests
impacts more peoples workflow

**Comment by pedromartinez** [comment]
Agreed!

**Comment by grosser** [comment]
done

**Comment by grosser** [comment]
should be fixed in 1.21.0
